python_modules/dagster/dagster/_core/definitions/asset_dep.py

```diff
@@ -66,3 +77,10 @@ def __new__(
                 PartitionMapping,
             ),
         )
+
+    @staticmethod
+    def from_coercible(arg: "CoercibleToAssetDep"):
+        if isinstance(arg, AssetDep):
```

Comment:
could be a one-liner 
https://github.com/dagster-io/dagster/pull/16432#discussion_r1323533856